idaes/core/util/model_diagnostics.py

```diff
@@ -694,10 +713,152 @@ def find_irreducible_degenerate_sets(self, verbose=True, tee=False):
     # Note: This makes sense as a static method
     @staticmethod
     def print_variable_bounds(v):
-        """Print variable, bounds, and value
+        """
+        Print variable, bounds, and value
+
         Argument:
             v: variable
+
         Return:
             nothing
+
         """
         print(v, "\t\t", v.lb, "\t", v.value, "\t", v.ub)
+
+
+def get_valid_range_of_component(component):
+    """
+    Return the valid range for a component as specified in the model metadata.
+
+    Args:
+        component - Pyomo component to get valid range for
+
+    Returns:
+        valid range for component if found. This will either be a 2-tuple (low, high) or None.
+
+    Raises:
+        AttributeError if metadata object not found
+
+    """
+    # Get metadata for component
+    parent = component.parent_block()
+
+    try:
+        if hasattr(parent, "params"):
+            meta = parent.params.get_metadata().properties
+        else:
+            meta = parent.get_metadata().properties
+    except AttributeError:
+        raise AttributeError(f"Could not find metadata for component {component.name}")
+
+    # Get valid range from metadata
+    try:
+        n, i = meta.get_name_and_index(component.parent_component().local_name)
+        cmeta = getattr(meta, n)[i]
+        valid_range = cmeta.valid_range
+    except ValueError:
+        # Assume no metadata for this property
+        _log.debug(f"No metadata entry for component {component.name}; returning None")
+        valid_range = None
+
+    return valid_range
+
+
+def set_bounds_from_valid_range(component, descend_into=True):
+    """
+    Set bounds on Pyomo components based on valid range recorded in model metadata.
+    WARNING - this function will overwrite any bounds already set on the component/model.
+
+    This function will iterate over component data objects in Blocks and indexed components.
+
+    Args:
+        component - Pyomo component to set bounds on. This can be a Block, Var or Param.
+        descend_into - (optional) Whether to descend into components on child Blocks (default=True)
```

Comment:
Change, for example, the documentation for `component` to:
```
        component : Pyomo component to set bounds on. This can be a Block, Var or Param.
```
(i.e. replace the `-` with a `:`). Similarly for `descend_into`.